lib/web/fetch/util.js

```diff
@@ -57,6 +60,16 @@ function responseLocationURL (response, requestFragment) {
   return location
 }
 
+/**
+ * If string contains non-ASCII characters, assumes it's UTF-8 encoded and decodes it.
+ * Since UTF-8 is a superset of ASCII, this will work for ASCII strings as well.
+ * @param {string} value
+ * @returns {string}
+ */
+function normalizeBinaryStringToUtf8 (value) {
+  return Buffer.from(value, 'binary').toString('utf8')
+}
```

Comment:
```

```C++
  let b = Buffer.from(value, 'binary');
  if(buffer.isAscii(b)) { return value; } // skip new string alloc.
  return b.toString('utf8'); // must alloc a new string
```

Reply:
No need to call `Buffer.from()`. `Buffer.isAscii(b)` is a static function.

Reply:
@anonrig We get at the core of the issue here. The `value` is a string not a buffer. At some point it was wrongly interpreted as latin1 (maybe in `http`) and so we must now convert it to a buffer and then back to a string.


I think that this function should receive a Buffer and then the Buffer could be interpreted as UTF-8 and we'd be done.

Reply:
But this is far beyond the scope of this PR.

Reply:
The irony is that we seem to try to purposefully (in http) prevent people from sending UTF-8 content as UTF-8... and then we go back and say "no, it is really UTF-8".

Much technical debt.